satpy/readers/ahi_hsd.py

```diff
@@ -487,6 +487,8 @@ def _vis_calibrate(self, data):
 
     def _ir_calibrate(self, data):
         """IR calibration."""
+        # No radiance -> no temperature
+        data = da.where(data == 0, np.float32(np.nan), data)
```

Comment:
you can use `fillna` which should preserve the type.
```suggestion
        data = data.fillna(np.nan)
```

Reply:
That was stupid, forget that